### PR TITLE
Rename DIET methods

### LINKED ISSUE #8671: API Review: Rename DIET methods
FindDelegatedIdentityEntityType -> FindEntityType
GetDelegatedIdentityEntityTypes -> GetEntityTypes
IsDelegatedIdentityEntityType -> HasDefiningNavigation
AddDelegatedIdentityEntityType -> AddEntityType
RemoveDelegatedIdentityEntityType -> RemoveEntityType

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #8671

### CODE REVIEW

File: src/EFCore/ChangeTracking/Internal/StateManager.cs [hunk 1]
```diff
@@ -136,7 +136,7 @@ public virtual InternalEntityEntry GetOrCreateEntry(object entity)
 
                 if (entityType == null)
                 {
-                    if (_model.IsDelegatedIdentityEntityType(entity.GetType()))
+                    if (_model.HasEntityTypeWithDefiningNavigation(entity.GetType()))
```
Comment (divega) [inline src/EFCore/ChangeTracking/Internal/StateManager.cs]: Should this be `IsEntityTypeWithDefiningNavigation`?

Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/StateManager.cs]: It looks strange to me since it's called on the model.

Comment (smitpatel) [inline src/EFCore/ChangeTracking/Internal/StateManager.cs]: `Has` sounds more appropriate since it is asking model - do you have ~DIET~ with this type

Comment (divega) [inline src/EFCore/ChangeTracking/Internal/StateManager.cs]: Ok

**Pull request closed without merging**